ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs

```diff
@@ -2508,6 +2513,19 @@ prop_never_above_hardlimit serverAcc
                                      )
                     . property
                     $ incomingConns cmc <= fromIntegral hardlimit
+                (TrPruneConnections prunnedSet numberToPrune choiceSet) ->
```

Comment:
I'd prefer a separate test for this. It will make it easier to change the counters test to use the Signal API and the pruning test doesn't require it.

Reply:
I thought about making a separate test but I decided against it because we have to be careful for the test suite to not take too much time to run.  This test runs in about 30sec, if we duplicate it will double the time.  Still I like to keep properties that test different thing separate as much as possible, whether this will be possible is not yet clear.   Maybe, we'll need to optimise `io-sim` at some point.   As for signal api, it might not be worth it in this test, the condition is precise and clear as it stands.

Reply:
Fair enough!